pkg/certificates/config_test.go

```diff
@@ -0,0 +1,433 @@
+			configYaml: `
+clusterName: test-cluster
+controlPlane:
+  nodes:
+  - 192.168.1.10
+  os: ubuntu
+  sshKey: /tmp/test-key
+  sshUser: ec2-user
+`,
+			expectError: false,
+		},
+		{
+			name: "invalid config - missing cluster name",
+			configYaml: `
+controlPlane:
+  nodes:
+  - 192.168.1.10
+  os: ubuntu
+  sshKey: /tmp/test-key
+  sshUser: ec2-user
+`,
+			expectError: true,
+		},
+		{
+			name: "invalid config - missing control plane nodes",
+			configYaml: `
+clusterName: test-cluster
+controlPlane:
+  os: ubuntu
+  sshKey: /tmp/test-key
+  sshUser: ec2-user
+`,
+			expectError: true,
+		},
+		{
+			name: "invalid config - unsupported OS",
+			configYaml: `
+clusterName: test-cluster
+controlPlane:
+  nodes:
+  - 192.168.1.10
+  os: windows
+  sshKey: /tmp/test-key
+  sshUser: ec2-user
+`,
+			expectError: true,
+			errorMsg:    "unsupported OS",
+		},
+		{
+			name: "valid config with SSH password",
+			configYaml: `
+clusterName: test-cluster
+controlPlane:
+  nodes:
+  - 192.168.1.10
+  os: ubuntu
+  sshKey: /tmp/test-key
+  sshUser: ec2-user
+  sshPasswd: password123
+`,
+			expectError: false,
+		},
+		{
+			name: "valid config with redhat OS",
+			configYaml: `
+clusterName: test-cluster
+controlPlane:
+  nodes:
+  - 192.168.1.10
+  os: redhat
+  sshKey: /tmp/test-key
+  sshUser: ec2-user
+`,
+			expectError: false,
+		},
+		{
+			name: "valid config with bottlerocket OS",
+			configYaml: `
+clusterName: test-cluster
+controlPlane:
+  nodes:
+  - 192.168.1.10
+  os: bottlerocket
+  sshKey: /tmp/test-key
+  sshUser: ec2-user
+`,
+			expectError: false,
+		},
+		{
+			name: "valid config with different OS types",
+			configYaml: `
+clusterName: test-cluster
+controlPlane:
+  nodes:
+  - 192.168.1.10
+  os: ubuntu
+  sshKey: /tmp/test-key
+  sshUser: ec2-user
+etcd:
+  nodes:
+  - 192.168.1.20
+  os: bottlerocket
+  sshKey: /tmp/test-key
+  sshUser: ec2-user
+`,
+			expectError: false,
+		},
+	}
+}
+
+// TestParseConfig tests the ParseConfig function.
+func TestParseConfig(t *testing.T) {
+	tests := getParseConfigTestCases()
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			runParseConfigTest(t, tt)
+		})
+	}
+}
+
+// validateNodeConfigTestCase defines a test case for validateNodeConfig.
+type validateNodeConfigTestCase struct {
+	name        string
+	config      NodeConfig
+	component   string
+	expectError bool
+}
+
+// runValidateNodeConfigTest runs a single validateNodeConfig test case.
+func runValidateNodeConfigTest(t *testing.T, tt validateNodeConfigTestCase) {
+	cleanup := setupSSHKeyForTest(t, tt.config.SSHKey)
+	defer cleanup()
+
+	err := validateNodeConfig(&tt.config)
+	if tt.expectError && err == nil {
+		t.Error("expected error but got none")
+	}
+	if !tt.expectError && err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+}
+
+// getValidateNodeConfigTestCases returns test cases for validateNodeConfig.
+func getValidateNodeConfigTestCases() []validateNodeConfigTestCase {
```

Comment:
Sorry, I might be missing somehting but didn't we validate it here TestValidateConfig?